packages/plugin-ext-headless/src/main/node/message-registry-main.ts

```diff
@@ -0,0 +1,49 @@
+// *****************************************************************************
+// Copyright (C) 2023 EclipseSource and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { interfaces } from '@theia/core/shared/inversify';
+import { MessageService } from '@theia/core/lib/common/message-service';
+import { MessageRegistryMain, MainMessageType, MainMessageOptions, MainMessageItem } from '@theia/plugin-ext/lib/common/plugin-api-rpc';
+
+export class MessageRegistryMainImpl implements MessageRegistryMain {
```

Comment:
This also seems to be a 1:1 copy from the plugin-ext? Could we not simply extend their implementation? I know they are in browser/node respectively but we could move them to common. I see that they do have some dependencies to specific types that we do not use for headless so I'll leave it up to you whether you want to have another look at it.

Reply:
I saw that you refactored the code nicely into a `BasicMessageRegistryMainImpl`, works for me.